typescript/cli/src/read/core.ts

```diff
@@ -23,17 +28,46 @@ export async function executeCoreRead({
     );
   }
 
-  const evmCoreReader = new EvmCoreReader(context.multiProvider, chain);
-  try {
-    return evmCoreReader.deriveCoreConfig({
-      mailbox,
-      interchainAccountRouter: addresses?.interchainAccountRouter,
-    });
-  } catch (e: any) {
-    errorRed(
-      `❌ Failed to read core config for mailbox ${mailbox} on ${chain}:`,
-      e,
-    );
-    process.exit(1);
+  const protocolType = context.multiProvider.getProtocol(chain);
+
+  switch (protocolType) {
+    case ProtocolType.Ethereum: {
+      const evmCoreReader = new EvmCoreReader(context.multiProvider, chain);
+      try {
+        return evmCoreReader.deriveCoreConfig({
+          mailbox,
+          interchainAccountRouter: addresses?.interchainAccountRouter,
```

Comment:
ahhh just because of the ICA router we can't dedupe more logic

@xeno097 how critical is that in the core read? like who actually needs to check ICA router on EVMs as part of the core deployment?

Reply:
I can also create the same method signature for `deriveCoreConfig` in `CosmosNativeCoreReader` and simply ignore the ICA address there. But like in the reply below I would suggest creating an additional PR just for refactoring